src/Components/Task.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from 'react';
+
+const Task = ({ description, setTaskList, tasks, id, setTaskToBeEdited, setNewDescription }) => {
+    const [isStrikeThrough, setStrikeThrough] = useState(false);
+
+    const removeTask = (id) => {
+        return setTaskList(tasks.filter(task => task.id !== id))
+    };
+
+    useEffect(() => {
+        const modal = document.querySelector('.modal');
+        const overlay = document.querySelector('.overlay');
+        const editButtons = Array.from(document.querySelectorAll('.button__edit'));
+        
+        function showModal() {
+            modal.classList.remove('hidden');
+            overlay.classList.remove('hidden');
+        }
+
+        editButtons.forEach(editButton => editButton.addEventListener('click', showModal));
```

Comment:
If we want to add click listeners to edit buttons we can do so in the JSX in the `return` block. Please remove this line and handle this logic in JSX.